test/memory-usage/tail-big-log-file.js

```diff
@@ -0,0 +1,73 @@
+'use strict'
+
+const fs = require('fs')
+const os = require('os')
+const path = require('path')
+const readline = require('readline')
+const {spawn} = require('child_process')
+const {test} = require('tap')
+const TailFile = require('../../index.js')
+
+const ALL_DONE_AFTER_TIME = 5000 // Close tail after 5 seconds of inactivity
+const shellCmd = path.join(__dirname, 'populate-big-log.sh')
+
+test('Memory consumption', async (t) => {
+  const LOG_FILE_NAME = 'big-log-file.log'
+  const testDir = os.tmpdir()
+  const LOG_FILE_PATH = path.join(testDir, LOG_FILE_NAME)
+
+  t.test('Create a stub for the large log file', async (t) => {
+    await fs.promises.writeFile(LOG_FILE_PATH, '', 'utf8')
+  })
+
+  t.test('Report memory consumptions when lots of large lines are consumed', (t) => {
+    const tail = new TailFile(LOG_FILE_PATH)
+      .on('error', (err) => {
+        t.fail(err)
+      })
+
+    t.teardown(async () => {
+      await tail.quit()
+    })
+
+    const lineSplitter = readline.createInterface({
+      input: tail
+    })
+
+    let counter = 0
+    let timer = null
+    // Resident Set Size - occupied memory including heap, segment and stack
+    let rssMin = Infinity
+    let rssMax = 0
+
+    lineSplitter.on('line', (line) => {
+      clearTimeout(timer)
+      if (++counter % 10000 === 0) {
+        const usage = process.memoryUsage()
+        if (usage.rss < rssMin) rssMin = usage.rss
+        if (usage.rss > rssMax) rssMax = usage.rss
```

Comment:
NIT: We could use `Math.min()` and `Math.max()` instead :)

Reply:
sure